### ISSUE TITLE
Bug in SpringBootExstension class

### ISSUE THREAD

**Issue by Jonatha1983** [spring-projects/spring-boot#18997]
Hi 

Seems like you got a bug in the SpringBootExstension class:
```
private static String getArchiveBaseName(AbstractArchiveTask task) {
		try {
			Method method = findMethod(task.getClass(), "getArchiveBaseName");
			if (method != null) {
				return (String) method.invoke(task);
			}
		}
```

The method getArchiveBaseName return a Property<String> instance. 

So I suggest to change to change it to:
```
 private static String getArchiveBaseName(AbstractArchiveTask task) {
        try {
            Method method = findMethod(task.getClass(), "getArchiveBaseName");
            if (method != null) {
                Property<String> property = (Property<String>) method.invoke(task)
                return property.get()
            }
```
Thanks

**Comment by wilkinsona** [comment]
Well spotted. Thank you. I have this on a branch where I've been doing some work with Gradle 6 but need to apply it to master.

**Comment by wilkinsona** [comment]
As it hasn't been released yet, I've re-opened #18663 to make the necessary change.
